File: app/src/main/java/com/developer/valyutaapp/model/Valute.java
package com.developer.valyutaapp.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

import io.reactivex.annotations.NonNull;


@Entity(tableName = "valute")
@Root(name="Valute")
public class Valute {

    @NonNull
    @Attribute(name = "ID")
    @PrimaryKey(autoGenerate = false)
    private int id;

    @Element(name = "CharCode")
    public String charCode;

    @Element(name = "Nominal")
    public int nominal;

    @Element(name = "Name")
    public String name;

    @Element(name = "Value")
    public String value;

    public int sortValute;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCharCode() {
        return charCode;
    }

    public void setCharCode(String charCode) {
        this.charCode = charCode;
    }

    public int getNominal() {
        return nominal;
    }

    public void setNominal(int nominal) {
        this.nominal = nominal;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public int getSortValute() {
        return sortValute;
    }

    public void setSortValute(int sortValute) {
        this.sortValute = sortValute;
    }
}

File: app/src/main/java/com/developer/valyutaapp/domain/entities/Valute.kt
package com.developer.valyutaapp.domain.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Entity(tableName = "valute")
@Root(name = "Valute", strict = false)
data class Valute(

    @field:Attribute(name = "ID")
    @param:Attribute(name = "ID")
    @PrimaryKey(autoGenerate = false)
    var id: Int = 0,

    @field:Element(name = "CharCode")
    @param:Element(name = "CharCode")
    var charCode: String = "",

    @field:Element(name = "Nominal")
    @param:Element(name = "Nominal")
    var nominal: Int = 0,

    @field:Element(name = "Name")
    @param:Element(name = "Name")
    var name: String = "",

    @field:Element(name = "Value")
    @param:Element(name = "Value")
    var value: String = "",

    var dates: String = "",
    var favoritesValute: Int = 0,
    var favoritesConverter: Int = 0,
)